HIP/hip-336.md

```diff
@@ -0,0 +1,301 @@
+
+As a user, I want to modify or remove an allowance I have granted to another user so that I can manage that user's ability to spend my tokens over time.
+
+As a user that has been granted an allowance, I want to be able to  transfer tokens owned by the owner account as if they were my tokens so that I can issue transfers on their behalf without them having to expose their private key.
+
+As a user, I want to query my account for the current token allowances that I have granted to other users so that I can monitor the activity of my delegates.
+  
+## Specification
+
+### Terminology
+
+The following terms will be used repeatedly throughout this section:
+
+- owner: the account that owns the tokens.
+- spender: the delegate account, the account being granted the power to spend the owner’s tokens.
+- recipient: the receiver of tokens when the spender issues a `transferFrom` operation.
+
+### Add Allowance Balances to CryptoGetInfoResponse
+
+The repeated fields `allowances` and `tokenAllowances` will be added to `CryptoGetInfoResponse` message. The `allowances` and `tokenAllowances` fields will contain hbar and token (fungible and non-fungible) allowances respectively.
+
+```protobuf
+// existing protobuf
+message CryptoGetInfoResponse {
+	// new nested message
+	message CryptoAllowance {
+		AccountID spender = 1;
+		int64 allowance = 2;
+	}
+
+	// new nested message
+	message TokenAllowance {
+		TokenID tokenId = 1;
+		AccountID spender = 2;
+		google.protobuf.Int64Value allowance = 3;
+		int32 decimals = 4;
+		google.protobuf.BoolValue approvedForAll = 5;
+	}
+	
+	// existing nested message
+	message AccountInfo {
+		// existing fields
+		AccountID accountID = 1;
+		string contractAccountID = 2;
+		bool deleted = 3;
+		AccountID proxyAccountID = 4;
+		int64 proxyReceived = 6;
+		Key key = 7;
+		uint64 balance = 8;
+		uint64 generateSendRecordThreshold = 9 [deprecated=true];
+		uint64 generateReceiveRecordThreshold = 10 [deprecated=true];
+		bool receiverSigRequired = 11;
+		Timestamp expirationTime = 12;
+		Duration autoRenewPeriod = 13;
+		repeated LiveHash liveHashes = 14;
+		repeated TokenRelationship tokenRelationships = 15;
+		string memo = 16;
+		int64 ownedNfts = 17;
+		int32 max_automatic_token_associations = 18;
+
+		// new fields
+		int32 totalAllowances = 19;
+		repeated CryptoAllowance allowances = 20;
+		repeated TokenAllowance tokenAllowances = 21;
+	}
+
+	// existing fields
+	ResponseHeader header = 1;
+	AccountInfo accountInfo = 2;
+}
+```
+
+The `TokenAllowance` message will contain the allowance information for a single fungible or non-fungible token spendable by the `spender`.
+
+If the token type is non-fungible, the `approvedForAll` field may be set to `true` if the spender has been granted access to all instances of the NFT owned by the owner. Fungible tokens will not `approvedForAll` as this field is only relevant for NFTs.
```

Comment:
Grammar: "Fungible tokens will not `approvedForAll` as this field"

Reply:
We should also document that if approvedForAll is true, then `amount` (or `allowance`) will be 0.

Reply:
Fixed